api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -26,12 +26,22 @@
 import org.apache.iceberg.Transaction;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotFoundException;
 
 /**
  * A Catalog API for table create, drop, and load operations.
  */
 public interface Catalog {
 
+  /**
+   * Return all the identifiers under this namespace.
```

Comment:
All identifiers or all identifiers that are tables? I think we should consider how to handle catalogs that store views and tables in the same namespace.

Reply:
My concern is that this will bring in overheads. For example in Hive Catalog, we need to load each table to check the metadata one by one. I'm not sure if there's a Hive metastore API which could load in all the table details under the same DB.

Reply:
Aren't we calling `client.getAllTables(database)` to load tables in a specific database?

Reply:
AFAIK, `client.getAllTables` only returns all the table names in a specific db, if we want get table details, I think we need to load each table separately.

Reply:
Correct, I thought that's what we need to support `listTables` in Spark's `TableCatalog` as it also returns just an array of identifiers.